components/findMatch.tsx

```diff
@@ -0,0 +1,112 @@
+import React, {Component} from "react";
+import { Button, TextField, ToggleButton, ToggleButtonGroup } from "@mui/material";
+
+export class FindMatch extends Component {
+
+    selected : boolean = false;
+    id : string = "";
+
+    changeStatus = () => {
+        this.selected = !this.selected;
+        this.updateView();
+    }
+
+    updateView = () => {
+        const el = document.getElementById("search");
+        
+        if(el){
+            el.innerHTML  = this.selected ? "Cancel" : "Find match";
+        }
```

Comment:
Con React non dovresti andare a modificare il contenuto mostrato andando a modificare l'HTML. 
Meglio il [conditional rendering](https://it.reactjs.org/docs/conditional-rendering.html)